extensions/mongodb/src/main/java/com/hazelcast/jet/mongodb/MongoSourceBuilder.java

```diff
@@ -452,15 +475,26 @@ private Stream(
         }
 
         /**
-         * If {@code true}, the lack of database or collection will cause an error.
-         * If {@code false}, database and collection will be automatically created.
-         * Default value is {@code true}.
+         * If set to true, reading will be done in only one thread.
+         *
+         * @param forceReadParallelismOne if true, reading will be done in only one thread.
+         */
+        @Nonnull
+        public Stream<T> forceReadParallelismOne(boolean forceReadParallelismOne) {
```

Comment:
can't this method be in `Base` instead of being duplicated? If you need a specific return type, you could use `class Batch<T> extends Base<T, Batch<T>> `

Reply:
Yeah but some methods change the `<T>` inside; then referencing to the "self-type" won't be possible.

Reply:
you do not have to move everything to `Base`, you could keep `mapFn` or `collection` or others where they are (BTW, I can see very interesting abuse of generics in these methods). But common methods which do not change type could be in single place. 
If you want to fight with generics, you could define method and javadoc for `mapFn` or `collection` and then only use covariant return type, but as long as this will be 2 or 3 methods this would not make much difference.

Reply:
Right, I understand now. Changed :)